src/mantine-core/src/components/Modal/Modal.tsx

```diff
@@ -193,6 +193,7 @@ export function Modal(props: ModalProps) {
         {(transitionStyles) => (
           <Box className={cx(classes.root, className)} {...others}>
             <div
+              id={baseId}
```

Comment:
You are adding id to the inner element, not to the root. It should be added to the Box component above.

Reply:
:+1: updated.